src/state/pools/fetchPools.ts

```diff
@@ -69,3 +71,29 @@ export const fetchPoolsTotalStaking = async () => {
     })),
   ]
 }
+
+export const fetchPoolStakingLimit = async (sousId: number): Promise<BigNumber> => {
+  try {
+    const sousContract = getSouschefV2Contract(sousId)
+    const stakingLimit = await sousContract.methods.poolLimitPerUser().call()
+    return new BigNumber(stakingLimit)
+  } catch (error) {
+    return BIG_ZERO
+  }
+}
+
+export const fetchPoolsStakingLimits = async (): Promise<{ [key: string]: BigNumber }> => {
+  const validPools = poolsConfig.filter((p) => p.stakingToken.symbol !== 'BNB' && !p.isFinished)
+
+  // Get the staking limit for each valid pool
+  // Note: We cannot batch the calls via multicall because V1 pools do not have "poolLimitPerUser" and will throw an error
+  const stakingLimitPromises = validPools.map((validPool) => fetchPoolStakingLimit(validPool.sousId))
```

Comment:
So are we attempting to fetch a staking limit for every pool apart from BNB staking pools & finished pools?

Are these staking limits going to become ubiquitous, or will it only be for specific pools? If it's the latter... maybe we should assign a boolean to these `hasStakingLimit` pools within the pools config to help filter the number of calls we need to make here? Might it also then allow us to do a multicall as V1 pools' `hasStakingLimit` would be false/undefined?

Doing anything for _every_ pool gives me the fear because what we do once, we do 60+ times. How's the performance with 60+ calls?

Reply:
Yeah...I agree. I am hoping we move a more efficient design where we don't fetch everything at once.

> Might it also then allow us to do a multicall as V1 pools' hasStakingLimit would be false/undefined?
This might be the solution we have to take.